radix-engine-interface/src/blueprints/epoch_manager/invocations.rs

```diff
@@ -50,9 +51,62 @@ pub const EPOCH_MANAGER_NEXT_ROUND_IDENT: &str = "next_round";
 
 #[derive(Debug, Clone, Eq, PartialEq, Sbor)]
 pub struct EpochManagerNextRoundInput {
+    /// Current round number.
+    /// Please note that in case of liveness breaks, this number may be different than previous
+    /// reported `round + 1`. Such gaps are considered "round leader's fault" and are penalized
+    /// on emission, according to leader reliability statistics (see `LeaderProposalMissHistory`).
     pub round: u64,
+
+    /// A captured history of leader proposal misses since the previous reported round.
+    pub leader_proposal_misses: LeaderProposalMissHistory,
```

Comment:
Can we add a comment that we should change this at some point so that the engine itself knows the leaders who missed the proposals. (ie we shouldn't be relying on the consensus layer to tell us this otherwise a quorum of validators may potentially gang up on other validators).

Reply:
From discussing this a few days ago, even in that set-up, a byzantine quorum could just pretend that any validators they don't like missed their proposals anyway. So the attack surface differential in these two scenarios is rather small.

I agree it would be nice-to-have, but I'm not sure how important it is.

Reply:
I can add the comment, but more importantly I have this question:

Do we want [hard] to move the leader rotation management to the Engine? 

(I mean: are there maybe more reasons why it makes sense? Is this conceptually very right thing?)

If the answer is "yes", then I suggest we design it and impl it right now, in a separate refactor (before continuing the "Emissions" work).

Reply:
I got the answer:
<img width="564" alt="image" src="https://user-images.githubusercontent.com/119413677/236304500-4ea59208-5fc5-40fb-b4bb-8738c1dc63b0.png">

with this assumption, I guess I can add the requested TODO here and merge :)